backend/src/packages/chaiNNer_pytorch/pytorch/io/load_model.py

```diff
@@ -77,53 +73,25 @@ def load_model_node(path: str) -> tuple[PyTorchModel, str, str]:
 
     try:
         logger.debug(f"Reading state dict from path: {path}")
-        extension = os.path.splitext(path)[1].lower()
 
-        if extension == ".pt":
-            state_dict = torch.jit.load(  # type: ignore
-                path, map_location=pytorch_device
-            ).state_dict()
-        elif extension == ".pth":
-            state_dict = torch.load(
-                path,
-                map_location=pytorch_device,
-                pickle_module=RestrictedUnpickle,  # type: ignore
-            )
-        elif extension == ".ckpt":
-            checkpoint = torch.load(
-                path,
-                map_location=pytorch_device,
-                pickle_module=RestrictedUnpickle,  # type: ignore
-            )
-            if "state_dict" in checkpoint:
-                checkpoint = checkpoint["state_dict"]
-            state_dict = parse_ckpt_state_dict(checkpoint)
-        elif extension == ".safetensors":
-            state_dict = load_file(path, device=str(pytorch_device))
-            logger.info(state_dict.keys())
-        else:
-            raise ValueError(
-                f"Unsupported model file extension {extension}. Please try a supported model type."
-            )
-
-        model = load_state_dict(state_dict)
+        model_descriptor = ModelLoader(pytorch_device).load_from_file(path)
 
-        for _, v in model.named_parameters():
+        for _, v in model_descriptor.model.named_parameters():
             v.requires_grad = False
-        model.eval()
-        model = model.to(pytorch_device)
-        if not hasattr(model, "supports_fp16"):
-            model.supports_fp16 = False  # type: ignore
-        should_use_fp16 = exec_options.use_fp16 and model.supports_fp16
+        model_descriptor.model.eval()
+        model_descriptor = model_descriptor.to(pytorch_device)
+        if not hasattr(model_descriptor, "supports_fp16"):
+            model_descriptor.supports_fp16 = False  # type: ignore
```

Comment:
The fuck?

Reply:
nice catch. that's what i get for blindly replacing variable names